presto-main/src/main/java/com/facebook/presto/cost/ExchangeStatsRule.java

```diff
@@ -26,13 +26,14 @@
 import java.util.Optional;
 
 import static com.facebook.presto.cost.PlanNodeStatsEstimateMath.addStatsAndMaxDistinctValues;
+import static com.facebook.presto.sql.planner.plan.Patterns.exchange;
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.base.Verify.verify;
 
 public class ExchangeStatsRule
-        extends SimpleStatsRule
+        extends SimpleStatsRule<ExchangeNode>
 {
-    private static final Pattern<ExchangeNode> PATTERN = Pattern.typeOf(ExchangeNode.class);
+    private static final Pattern<ExchangeNode> PATTERN = exchange();
```

Comment:
You can also inline this, as it is in `JoinStatsRule` (might be second commit)

Reply:
Inlining will make it create new object on every call to `Rule#getPattern`. I think `JoinStatsRule` needs to be fixed to add a static member `PATTERN`.

Reply:
Obviously -- but there is only one call in the lifetime (this is a singleton), so no difference.

Reply:
@findepi I see. Everywhere else we use a static member variable. For consistency, let's use the same in `JoinStatsRule`.